packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -402,15 +404,50 @@ function startPollingElementFromPoint(): ThunkAction<() => void, State, unknown,
   }
 }
 
-export function initialize(): ThunkAction<() => void, State, unknown, Action> {
-  return dispatch => {
+export interface IMessageChannel {
+  postMessage(message: any, transferables?: Transferable[]): void
+  dispatchBuffered(): void
+}
+
+function registerBuilderComponents(state: State, messageChannel: IMessageChannel) {
```

Comment:
Factor out builder component registration into its own function, make it a part of the initialization sequence (see `initialize` below).